### ISSUE TITLE
WDE002924

### ISSUE THREAD

**Issue by jacobhallgren** [dresden-elektronik/deconz-rest-plugin#6141]
## Device

- Product name: WDE002924
- Manufacturer: Scheider Electric
- Model identifier: FLS/SYSTEM-M/4
- Device type : Please remove all unrelated device types. 
  - Switch

### Basic 
![basic](https://user-images.githubusercontent.com/25883555/174314284-49de0f73-086a-4c1d-8a1a-e033213bb2a0.PNG)

![basic](https://user-images.githubusercontent.com/25883555/174316249-9c0c1bed-ed84-4d7b-8824-281c173b5efa.PNG)

### Identify 
![identify](https://user-images.githubusercontent.com/25883555/174314324-992a9a4c-8d2f-449b-bd22-568d86a856b7.PNG)

<img width="507" alt="Screenshot 2022-06-17 at 20 58 28" src="https://user-images.githubusercontent.com/25883555/174386848-48882c09-c196-4441-a16b-5603a2f51c8a.png">
<img width="506" alt="Screenshot 2022-06-17 at 20 58 47" src="https://user-images.githubusercontent.com/25883555/174386861-ec3f1555-731a-4e50-b0f5-890f0750bd1d.png">
<img width="503" alt="Screenshot 2022-06-17 at 20 59 03" src="https://user-images.githubusercontent.com/25883555/174386863-eee6cd9c-517c-4ef0-bf9d-702811e40442.png">
<img width="510" alt="Screenshot 2022-06-17 at 20 59 18" src="https://user-images.githubusercontent.com/25883555/174386866-27cf3e1b-c90f-46ab-bca1-849ca91f0ed9.png">
<img width="499" alt="Screenshot 2022-06-17 at 20 59 33" src="https://user-images.githubusercontent.com/25883555/174386869-4f9c8fa1-76ab-47d7-87fa-b91a9a95c507.png">

### Other clusters that are not mentioned above 
![4 switches](https://user-images.githubusercontent.com/25883555/174314894-8008a6bb-3dc4-46d9-b94a-b9fd77447e69.PNG)

**Comment by Mimiix** [comment]
Missing screenshots of blue clusters. Please add them

**Comment by jacobhallgren** [comment]
> Missing screenshots of blue clusters. Please add them
<img width="507" alt="Screenshot 2022-06-17 at 20 58 28" src="https://user-images.githubusercontent.com/25883555/174386408-9fb8cf79-be2b-4682-bda1-af94502069a5.png">
<img width="506" alt="Screenshot 2022-06-17 at 20 58 47" src="https://user-images.githubusercontent.com/25883555/174386411-7faa4dd0-c357-4c68-9367-1eb0936750fc.png">
<img width="503" alt="Screenshot 2022-06-17 at 20 59 03" src="https://user-images.githubusercontent.com/25883555/174386413-f480e083-468f-4f86-b6f1-0bc930b22fc5.png">
<img width="510" alt="Screenshot 2022-06-17 at 20 59 18" src="https://user-images.githubusercontent.com/25883555/174386415-8bcf307a-ee29-4e0f-a135-585fbe147128.png">
<img width="499" alt="Screenshot 2022-06-17 at 20 59 33" src="https://user-images.githubusercontent.com/25883555/174386416-0278e23d-efcd-44b8-ae83-4029111ada6f.png">

**Comment by Mimiix** [comment]
Please add them to the original post.

**Comment by jacobhallgren** [comment]
> Please add them to the original post.

@Mimiix Done

**Comment by Mimiix** [comment]
Thanks :)

**Comment by jacobhallgren** [comment]
Any Update on tvOS?

**Comment by Mimiix** [comment]
@Smanar something you can do?

**Comment by Smanar** [comment]
Can make a try, but from my memory schneider devices have 'special trick", we need to use a specific endpoint on the coordinator, but I don't know wich one devices.

Can you try  this DDF

```
{
  "schema": "devcap1.schema.json",
  "manufacturername": "Scheider Electric",
  "modelid": "FLS/SYSTEM-M/4",
  "product": "4 gangs switch",
  "sleeper": true,
  "status": "Gold",
  "subdevices": [
    {
      "type": "$TYPE_SWITCH",
      "restapi": "/sensors",
      "uuid": [
        "$address.ext",
        "0x15",
        "0x0006"
      ],
      "items": [
        {
          "name": "attr/id"
        },
        {
          "name": "attr/lastannounced"
        },
        {
          "name": "attr/lastseen"
        },
        {
          "name": "attr/manufacturername"
        },
        {
          "name": "attr/modelid"
        },
        {
          "name": "attr/name"
        },
        {
          "name": "attr/swversion"
        },
        {
          "name": "attr/type"
        },
        {
          "name": "attr/uniqueid"
        },
        {
          "name": "config/battery"
        },
        {
          "name": "config/on"
        },
        {
          "name": "config/reachable"
        },
        {
          "name": "state/buttonevent"
        },
        {
          "name": "state/lastupdated"
        }
      ]
    }
  ],
  "bindings": [
    {
      "bind": "unicast",
      "src.ep": 21,
      "dst.ep": 1,
      "cl": "0x0006"
    },
    {
      "bind": "unicast",
      "src.ep": 22,
      "dst.ep": 1,
      "cl": "0x0006"
    },
    {
      "bind": "unicast",
      "src.ep": 23,
      "dst.ep": 1,
      "cl": "0x0006"
    },
    {
      "bind": "unicast",
      "src.ep": 24,
      "dst.ep": 1,
      "cl": "0x0006"
    }
  ]
}
```

It's a realy basic DDF, it just make the device includable in the API and configure bind for the cluster on/off for all 4 endpoints.

If it work the device will have a better name than 0x767C in deconz.
Then go in deconz, help / debug view, with flag "info" and "info_l2", and press button, if you are lucky you will see error about button map.

**Comment by jensflorian** [comment]
please reopen.
This device is also branded as Merten MEG5001-0300 (https://eref.se.com/de/de/scope/product-pdf/MEG5001-0300).
How do I force the above DDF example onto this device? I tried the hot reload option
<img width="1870" alt="Bildschirmfoto 2022-09-27 um 20 23 14" src="https://user-images.githubusercontent.com/58853392/192606365-c9370995-484f-41c8-b1f8-4bb38df7b8d4.png">

**Comment by Mimiix** [comment]
Sure.

**Comment by SwoopX** [comment]
@jensflorian You might want to type the correct manufacturer name 😋

**Comment by jensflorian** [comment]
Thank you very much, after correcting the typo, the device is also shown in Phoscon.
![E3C614F4-3AB4-42FC-B095-E6F150D47D2F](https://user-images.githubusercontent.com/58853392/192697072-cc6b2aa8-4ec3-442a-8c3a-8ecb4bcdd35b.jpeg)

Here is the output from help / debug view after pressing the top button:
![D93F94CB-0E0C-472E-8989-E3918BD1568A](https://user-images.githubusercontent.com/58853392/192697623-5a010470-d00a-4185-8157-a7e8ea97bcb4.jpeg)

 Can you create button maps from this?

**Comment by Smanar** [comment]
Nope thoses logs ^^.
Enable flag "info_l2", you will have thoses usefull error lines

```
	Line 4570:         DBG_Printf(DBG_INFO_L2, "[INFO] - No button map for: %s%s, endpoint: 0x%02X, cluster: %s, command: %s, payload: %s, zclSeq: %u\n",
	Line 5607:         DBG_Printf(DBG_INFO_L2, "[INFO] - No button handler for: %s%s, endpoint: 0x%02X, cluster: %s, command: %s, payload: %s, zclSeq: %u\n",
```

**Comment by jensflorian** [comment]
The above screenshot was with flag "info_l2" enabled. I have only the 1gang cover.
Top button:
`17:22:58:143 [INFO] - No button map for: FLS/SYSTEM-M/4, unicast to: 0x0000, endpoint: 0x15, cluster: ONOFF (0x0006), command: ON (0x01), payload: None, zclSeq: 33`

Bottom button:
`17:23:00:030 [INFO] - No button map for: FLS/SYSTEM-M/4, unicast to: 0x0000, endpoint: 0x15, cluster: ONOFF (0x0006), command: OFF (0x00), payload: None, zclSeq: 34`

This matches the AIRLINK device. So I added the device in button_maps.json as suggested by @Smanar and restarted deConz.

Now the log file returns after pressing the button:

`17:40:48:794 Websocket 192.168.1.130:36346 send message: {"attr":{"id":"113","lastannounced":null,"lastseen":"2022-09-28T15:40Z","manufacturername":"Schneider Electric","modelid":"FLS/SYSTEM-M/4","name":"Switch 113","swversion":"002.005.030 R","type":"ZHASwitch","uniqueid":"04:cd:15:ff:fe:c0:d1:14-ff-0006"},"e":"changed","id":"113","r":"sensors","t":"event","uniqueid":"04:cd:15:ff:fe:c0:d1:14-ff-0006"} (ret = -1093345760)
17:40:48:795 Websocket 192.168.1.130:36348 send message: {"attr":{"id":"113","lastannounced":null,"lastseen":"2022-09-28T15:40Z","manufacturername":"Schneider Electric","modelid":"FLS/SYSTEM-M/4","name":"Switch 113","swversion":"002.005.030 R","type":"ZHASwitch","uniqueid":"04:cd:15:ff:fe:c0:d1:14-ff-0006"},"e":"changed","id":"113","r":"sensors","t":"event","uniqueid":"04:cd:15:ff:fe:c0:d1:14-ff-0006"} (ret = -1093345760)
17:40:48:796 Websocket 192.168.1.130:36366 send message: {"attr":{"id":"113","lastannounced":null,"lastseen":"2022-09-28T15:40Z","manufacturername":"Schneider Electric","modelid":"FLS/SYSTEM-M/4","name":"Switch 113","swversion":"002.005.030 R","type":"ZHASwitch","uniqueid":"04:cd:15:ff:fe:c0:d1:14-ff-0006"},"e":"changed","id":"113","r":"sensors","t":"event","uniqueid":"04:cd:15:ff:fe:c0:d1:14-ff-0006"} (ret = -1093345760)
17:40:48:797 Websocket 192.168.1.130:36368 send message: {"attr":{"id":"113","lastannounced":null,"lastseen":"2022-09-28T15:40Z","manufacturername":"Schneider Electric","modelid":"FLS/SYSTEM-M/4","name":"Switch 113","swversion":"002.005.030 R","type":"ZHASwitch","uniqueid":"04:cd:15:ff:fe:c0:d1:14-ff-0006"},"e":"changed","id":"113","r":"sensors","t":"event","uniqueid":"04:cd:15:ff:fe:c0:d1:14-ff-0006"} (ret = -1093345760)
17:40:48:800 Websocket 192.168.1.130:36346 send message: {"e":"changed","id":"113","r":"sensors","state":{"buttonevent":1002,"lastupdated":"2022-09-28T15:40:48.793"},"t":"event","uniqueid":"04:cd:15:ff:fe:c0:d1:14-ff-0006"} (ret = -1093345760)
17:40:48:801 Websocket 192.168.1.130:36348 send message: {"e":"changed","id":"113","r":"sensors","state":{"buttonevent":1002,"lastupdated":"2022-09-28T15:40:48.793"},"t":"event","uniqueid":"04:cd:15:ff:fe:c0:d1:14-ff-0006"} (ret = -1093345760)
17:40:48:802 Websocket 192.168.1.130:36366 send message: {"e":"changed","id":"113","r":"sensors","state":{"buttonevent":1002,"lastupdated":"2022-09-28T15:40:48.793"},"t":"event","uniqueid":"04:cd:15:ff:fe:c0:d1:14-ff-0006"} (ret = -1093345760)
17:40:48:804 Websocket 192.168.1.130:36368 send message: {"e":"changed","id":"113","r":"sensors","state":{"buttonevent":1002,"lastupdated":"2022-09-28T15:40:48.793"},"t":"event","uniqueid":"04:cd:15:ff:fe:c0:d1:14-ff-0006"} (ret = -1093345760)
1`

The bindings tab in the DDF editor is still empty. Do I need to add anything here?

**Comment by Smanar** [comment]
>192.168.1.130:36346 send message: {"e":"changed","id":"113","r":"sensors","state":{"buttonevent":1002,"lastupdated":"2022-09-28T15:40:48.793"},"t":"event","uniqueid":"04:cd:15:ff:fe:c0:d1:14-ff-0006"} (ret = -1093345760) 17:40:48:801 Websocket

This one have worked, we can see `"buttonevent":1002` as the value is not null, it have worked at a moment.
You can still use the debug with "info_l2" (only, less talkative), it will be more visible.

>The bindings tab in the DDF editor is still empty

Not normal, you need to have thoses binding 
```
  "bindings": [
    {
      "bind": "unicast",
      "src.ep": 21,
      "dst.ep": 1,
      "cl": "0x0006"
    },
    {
      "bind": "unicast",
      "src.ep": 22,
      "dst.ep": 1,
      "cl": "0x0006"
    },
    {
      "bind": "unicast",
      "src.ep": 23,
      "dst.ep": 1,
      "cl": "0x0006"
    },
    {
      "bind": "unicast",
      "src.ep": 24,
      "dst.ep": 1,
      "cl": "0x0006"
```

You can see them in the DDF.
I making a bind (only for the cluster 0x0006 to test) but for all endpoint (If I m right you have 1 endpoint by button). 0x15 is the hexa version, 21 the decimal value.

If you haven't log "[INFO] - No button map for" for the second button, it's probably a problem with bind yes.

If you have them but no change on buttonevent return it s probably due to the button map file (The code use only endpoint 0x15 and 0x16, you have 2 mores on your device from capture)

It's a 4 gangs device ?

**Comment by jensflorian** [comment]
> Not normal, you need to have thoses binding

Sorry, my fault, I forgot to reload the json file after restarting deConz. Now I have all four endpoints in the DDF. Thanks!

>If you haven't log "[INFO] - No button map for" for the second button, it's probably a problem with bind yes.

The second button has buttonevent 2002. 

>It's a 4 gangs device ?

Depending on the button cover you are using its either  1gang(2button) or a 2gang(4button) switch. This will be commissioned during the pairing process, see https://www.productinfo.schneider-electric.com/wiser_home/sysm_sysd-connected-wireless-switch-1-gang_2-gang_device-user-guide_wiser-home/English/SysM_SysD%20-%20Connected%20Wireless%20Switch%201-gang_2-gang_Wiser_Home_Device%20user%20guide_0000774577.xml/$/WHM-SysM_SysD-Connectedwirelessswitch1-gang_2-gang-PairingthedeviceTSK_0000735364
As I have only the two button cover, I'm getting only button events 1002 and 2002.

I think this information is stored in the Merten specific cluster FF17. To read the clusters you need to wake up the device by pressing the top button 3x quickly. The green light will flash and you can read all the clusters:
<img width="776" alt="Bildschirmfoto 2022-09-29 um 20 29 21" src="https://user-images.githubusercontent.com/58853392/193116163-6aecd446-d15f-414b-82f7-a4f57c290aff.png">

Can you make a PR for your DDF and button_maps.json ? I'd be happy to have this device included in future deconz releases.

**Comment by Smanar** [comment]
Sure I can, but it's finished ?
Your device is only a on/off, no dimmer feature ?
If you have the dimmer feature (generaly long press) I thinkj you need to bind the cluster 0x0008 too with adding

```
    {
      "bind": "unicast",
      "src.ep": 21,
      "dst.ep": 1,
      "cl": "0x0008"
    },
    {
      "bind": "unicast",
      "src.ep": 22,
      "dst.ep": 1,
      "cl": "0x0008"
    },
    {
      "bind": "unicast",
      "src.ep": 23,
      "dst.ep": 1,
      "cl": "0x0008"
    },
    {
      "bind": "unicast",
      "src.ep": 24,
      "dst.ep": 1,
      "cl": "0x0008"
    }
```

> Pressed button 1 and 4 for about 5 seconds to enable individual buttons. :-)

BTW, you have the different value the attribute 0x0020 can have ?

**Comment by jensflorian** [comment]
> Your device is only a on/off, no dimmer feature ?

According to the manufacturer manual it is capable to dim, use a shutter and trigger a moment. However I do not get any other messages with "info_l2" enabled as for button events 1002 and 2002. There is no other message or error after long press. So, it needs probably to be configured by the manufacturer app and this is probably in the Merten specific cluster FF17 somewhere. I haven't figured this out yet.

>BTW, you have the different value the attribute 0x0020 can have ?

For endpoint 15 its "0" for the other 16,17 and 18 it's "1"

The attribute 0x0001 is "unknown for endpoint 15 and for the other 16,17 and 18 it's "7f"

**Comment by Smanar** [comment]
> For endpoint 15 its "0" for the other 16,17 and 18 it's "1"
> The attribute 0x0001 is "unknown for endpoint 15 and for the other 16,17 and 18 it's "7f"

Ok so need investigation, because I don't see the logic

>There is no other message or error after long press. So, it needs probably to be configured by the manufacturer app and this is probably in the Merten specific cluster FF17 somewhere. I haven't figured this out yet.

Or they are using the cluster 0x0008 , the level control, try to add them or can just bind one manualy to test.

Else I can make the PR as it ?

**Comment by jensflorian** [comment]
> Or they are using the cluster 0x0008 , the level control, try to add them or can just bind one manualy to test.

Tested it without success, is probably not enabled by default.

>Else I can make the PR as it ?

Don't forget to correct the typo in "manufacturername": "Scheider Electric" to Schneider.
Thank you for your input and help!

**Comment by Smanar** [comment]
Ok so have removed bind for cluster 0x0008

Final DDF https://github.com/dresden-elektronik/deconz-rest-plugin/pull/6373

**Comment by Smanar** [comment]
Oups, it' perhaps possible to mix 2 DDF, can you try this one ?

It's an update of this one devices\wiser\fuga_4button_battery_switch.json

```
{
  "schema": "devcap1.schema.json",
  "manufacturername": {"Schneider Electric", "Schneider Electric"],
  "modelid": ["FLS/AIRLINK/4", "FLS/SYSTEM-M/4"],
  "product": "LK Fuga 4 Button Switch",
  "sleeper": true,
  "status": "Gold",
  "subdevices": [
    {
      "type": "$TYPE_SWITCH",
      "restapi": "/sensors",
      "uuid": [
        "$address.ext",
        "0x15",
        "0x0006"
      ],
      "fingerprint": { "profile": "0x0104", "device": "0x000c", "endpoint": "0x15", "in": ["0x0000", "0x0001", "0x0006", "0x0008"] },
      "items": [
        {
            "name": "config/battery",
            "parse": {"ep": 21, "cl": "0x0001", "at": "0x0021", "eval": "Item.val = Attr.val / 2"},
            "awake": true
        },
        {
          "name": "attr/id"
        },
        {
          "name": "attr/lastannounced"
        },
        {
          "name": "attr/lastseen"
        },
        {
          "name": "attr/manufacturername"
        },
        {
          "name": "attr/modelid"
        },
        {
          "name": "attr/name"
        },
        {
          "name": "attr/swversion"
        },
        {
          "name": "attr/type"
        },
        {
          "name": "attr/uniqueid"
        },
        {
          "name": "config/on"
        },
        {
          "name": "config/reachable"
        },
        {
          "name": "state/buttonevent"
        },
        {
          "name": "state/lastupdated"
        }
      ]
    }
  ],
  "bindings": [
    {
      "bind": "unicast",
      "src.ep": 21,
      "cl": "0x0006"
    },
    {
      "bind": "unicast",
      "src.ep": 22,
      "cl": "0x0006"
    },
    {
      "bind": "unicast",
      "src.ep": 21,
      "cl": "0x0008"
    },
    {
      "bind": "unicast",
      "src.ep": 22,
      "cl": "0x0008"
    },
    {
      "bind": "unicast",
      "src.ep": 21,
      "cl": "0x0001",
      "report": [ {"at": "0x0021", "dt": "0x20", "min": 300, "max": 3600, "change": "0x01" } ]
    }
  ]
}
```

**Comment by jensflorian** [comment]
> possible to mix 2 DDF?
unfortunately not yet. This one doesn't work.  This will cause:

`  "modelid": "FLS/SYSTEM-M/4",
  "product": "FLS/SYSTEM-M/4",`
and endpoints missing.

**Comment by Smanar** [comment]
Oups my bad, have made a typo, have corrected the DDf on the previous post.

**Comment by jensflorian** [comment]
This one works!

**Comment by Smanar** [comment]
Nice, I will make a new PR with both device mixed, as it not possible to know if it's a 2 gang or 4 gang (same firmware, same hardware, just a setting) I don't think it can cause issue.

**Comment by frehov** [comment]
I'm very sorry to necro this issue, but if you have any trouble with the FF17 cluster, please take a look at the zigbee implementation docs from Schneider, which details the inner workings of the FF17 cluster for the these switches/remotes.
https://drive.google.com/file/d/1FANEmbeh89I6Hn9e6ddt1YzSzk_GofJd/view?usp=sharing

This device can function in 1-gang, 2-gang and 4-pushbutton mode, depending on FF17 cluster values.

These documents were found on an aws cloudfront distribution that had been indexed by duckduckgo, along with 7 other zigbee implementation documents https://drive.google.com/drive/folders/1PaPQfbuMHdtiV199EIIirJE5RbujuJXn?usp=sharing for different Schneider devices.

**Comment by Smanar** [comment]
Ha nice, usefull informations, thx
